client/src/components/SignOutButton.jsx

```diff
@@ -0,0 +1,80 @@
+import { Box, Button, Modal, Stack, Typography } from '@mui/material';
+import { signOut } from 'firebase/auth';
+import React, { useContext, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import AuthContext from '../contexts/AuthContext';
+import { auth } from '../utils/firebaseConfig';
+
+const SignOutButton = () => {
+  let navigate = useNavigate();
+  const [, setAuthUser] = useContext(AuthContext);
+  const [openModal, setOpenModal] = useState(false);
+  const handleSignOut = () => {
+    signOut(auth)
+      .then(() => {
+        // Sign-out successful.
+        setAuthUser(null);
+        navigate('/signin');
+      })
+      .catch((error) => {
+        // An error happened.
+      });
+  };
+  const handleOpenModal = () => setOpenModal(true);
+  const handleCloseModal = () => setOpenModal(false);
+  return (
+    <>
+      <Button
+        variant="rounded"
+        color="thai"
+        onClick={handleOpenModal}
+        sx={{
```

Comment:
This is okay for now and it probably nit, but I prefer styles into their own file instead of coupled with the component itself. That way we can re-use styles for similar components.

Again, not a big deal, just a nit.